x/auth/types/account.go

```
@@ -64,20 +72,32 @@ func (acc *BaseAccount) SetAddress(addr sdk.AccAddress) error {
 
 // GetPubKey - Implements sdk.AccountI.
 func (acc BaseAccount) GetPubKey() (pk crypto.PubKey) {
-	if len(acc.PubKey) == 0 {
+	if acc.PubKey == nil {
 		return nil
 	}
-
-	amino.MustUnmarshalBinaryBare(acc.PubKey, &pk)
-	return pk
+	content, ok := acc.PubKey.GetCachedValue().(crypto.PubKey)
+	if !ok {
+		return nil
+	}
+	return content
 }
 
 // SetPubKey - Implements sdk.AccountI.
 func (acc *BaseAccount) SetPubKey(pubKey crypto.PubKey) error {
 	if pubKey == nil {
 		acc.PubKey = nil
 	} else {
-		acc.PubKey = amino.MustMarshalBinaryBare(pubKey)
+		protoMsg, ok := pubKey.(proto.Message)
+		if !ok {
+			return sdkerrors.ErrInvalidPubKey
+		}
+
+		any, err := codectypes.NewAnyWithValue(protoMsg)
+		if err != nil {
+			return nil
```

why do we return `nil`?

---

good catch, we should return the error